src/blockMonitor/blockProcessor.ts

```diff
@@ -185,17 +199,32 @@ export class BlockProcessor<TBlock extends IBlockStub> extends StartStopService
             return await this.getBlockRemote(blockHash);
         }
     }
+
     // Processes a new block, adding it to the cache and emitting the appropriate events
     // It is called for each new block received, but also at startup (during startInternal).
     private async processBlockNumber(blockNumber: number) {
         try {
-            const observedBlock = await this.getBlockRemote(blockNumber);
+            // we cant process blocks greater than max depth of the cache
+            // so if the cache is empty any block is fine, otherwise
+            // the blocknumber cannot be more than maxDepth greater than the head
+            const maxBlock = this.mBlockCache.isEmpty // prettier-ignore
```

Comment:
While it does not matter logically, it is still better to split the remaining blocks to be added in chunks, so that the number of blocks kept unattahed in the BlockCache is bounded. Otherwise _all_ the blocks from the stored snapshot to the current head are kept in memory until they can be attached, since they can't be pruned before that.